### ISSUE TITLE
bug(datepicker): It is not announcing the date value on all calendars

### ISSUE THREAD

**Issue by zarend** [angular/components#23480]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the "basic datepicker" demo
3. Activate "Open calendar" button from any section, say, "Single date selection"
4. Focus on any date value
 
 
#### Expected Behavior
It should announce the date value, when focus on any date on the calendar

#### Actual Behavior
It is not announcing the date value, but announcing the row and column values. For example, it is announcing as "08 row 3 column 5" without actual date

#### Environment
OS| Browser | Screen reader
ChromeOS | Google Chrome 91.0.4472.167 | ChromeVox
Windows 10 | Google Chrome 92.0.4515.131 | NVDA/JAWS
Windows 10 | Firefox 90.0.2 (64-bit) | NVDA

##### Additional Notes
This is copied from an [internal bug report](http://b/195600261) found during a11y testing.

**Comment by zarend** [comment]
I can't seem to repro this anymore, maybe it already got fixed 🤔 .

Windows 10 Enterprise 10H2
Chrome 96.0.4664.110 
NVDA version 2021.3.1
JAWS version 2021.2103.174 ILM
